src/main/kotlin/com/ampnet/identityservice/service/impl/PinataServiceImpl.kt

```diff
@@ -0,0 +1,79 @@
+package com.ampnet.identityservice.service.impl
+
+import com.ampnet.identityservice.config.ApplicationProperties
+import com.ampnet.identityservice.exception.ErrorCode
+import com.ampnet.identityservice.exception.InternalException
+import com.ampnet.identityservice.service.PinataService
+import com.fasterxml.jackson.annotation.JsonProperty
+import mu.KLogging
+import org.springframework.http.HttpEntity
+import org.springframework.http.HttpHeaders
+import org.springframework.http.MediaType
+import org.springframework.stereotype.Service
+import org.springframework.web.client.RestClientException
+import org.springframework.web.client.RestTemplate
+import org.springframework.web.client.postForEntity
+
+@Service
+class PinataServiceImpl(
+    private val applicationProperties: ApplicationProperties,
+    private val restTemplate: RestTemplate
+) : PinataService {
+
+    companion object : KLogging()
+
+    private val pinataUrl = "https://api.pinata.cloud/users/generateApiKey"
+
+    @Throws(InternalException::class)
+    override fun getUserJwt(address: String): PinataResponse {
+        val request = PinataApiKeyRequest(address, maxUses = applicationProperties.pinata.maxUses)
+        try {
+            val httpEntity = HttpEntity(request, generateHeaders())
+            val response = restTemplate.postForEntity<PinataResponse>(pinataUrl, httpEntity)
+            if (response.statusCode.is2xxSuccessful) {
+                return response.body ?: throw InternalException(
+                    ErrorCode.USER_PINATA_JWT,
+                    "Missing body for successful response"
+                )
+            } else {
+                throw InternalException(
+                    ErrorCode.USER_PINATA_JWT,
+                    "Unsuccessful request to get Pinata JWT: ${response.statusCode}"
+                )
+            }
+        } catch (ex: RestClientException) {
+            throw InternalException(ErrorCode.USER_PINATA_JWT, "Failed to get Pinata JWT")
+        }
+    }
+
+    private fun generateHeaders(): HttpHeaders {
+        val headers = HttpHeaders()
+        headers.set(HttpHeaders.CONTENT_TYPE, MediaType.APPLICATION_JSON_VALUE)
+        headers.set(HttpHeaders.AUTHORIZATION, "Bearer ${applicationProperties.pinata.jwt}")
```

Comment:
Couldn't find this in the docs, as I see in the examples headers contain 
```headers: {
                pinata_api_key: pinataApiKey,
                pinata_secret_api_key: pinataSecretApiKey
}```

Hope you checked it with a real call to the service!

Reply:
You can use either API key + secret API key or just JWT as in this case.